from_mwt_ds/DataScience/vw_executor/vw.py

```diff
@@ -47,34 +48,50 @@ class _VwBin(_VwCore):
     def __init__(self, path: Path):
         super().__init__(path)
 
-    def run(self, args: str) -> str:
+    def run(self, args: str, out_path: Path) -> str:
         command = f'{self.path} {args}'
+        stdout_file = open(out_path.parent / (out_path.name + '.out.txt'), 'w')
+        stderr_temp = out_path.parent / (out_path.name + '.pending')
+        stderr_file = open(stderr_temp, 'w')
+
         process = subprocess.Popen(
             command.split(),
             universal_newlines=True,
             encoding='utf-8',
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE
+            stdout=stdout_file,
+            stderr=stderr_file
         )
-        error = process.communicate()[1]
-        return error
 
+        returncode = process.wait()
+        stdout_file.close()
+        stderr_file.close()
+
+        if returncode == 0:
+            os.rename(stderr_temp, out_path)
```

Comment:
Vw(... reset=True ....) is supposed to ignore and overwrite existing cache, so there is no guarantee that out_path doesn't exist here. And os.rename is failing on existing dst file

Reply:
looks like os.replace is better here?